gson/src/test/java/com/google/gson/functional/CircularReferenceTest.java

```diff
@@ -96,7 +97,15 @@ public JsonElement serialize(
             .create();
 
     // Circular reference to self can not be serialized
-    assertThrows(StackOverflowError.class, () -> gson.toJson(obj));
+    assertThrowsStackOverflow(() -> gson.toJson(obj));
+  }
+
+  /** Asserts that a {@link StackOverflowError} is thrown. */
```

Comment:
But it doesn't assert that, does it? 🙂  I think we can probably assume that `StackOverflowError` is at least in the cause chain, and probably that it is the [root cause](https://javadoc.io/static/com.google.guava/guava/33.4.0-jre/com/google/common/base/Throwables.html#getRootCause(java.lang.Throwable)). So then this?

```java
Throwable t = assertThrows(Throwable.class, runnable);
assertThat(Throwables.getRootCause(t)).isInstanceOf(StackOverflowError.class);
```

Reply:
> But it doesn't assert that, does it?

Yes you are right. My intention was to indicate that this methods is intended for cases where a `StackOverflowError` is expected, but then the implementation admits that it cannot reliably detect that.

----

> I think we can probably assume that `StackOverflowError` is at least in the cause chain

Yes that could be an option. Let's just hope that there is no JDK code which sets it only as 'suppressed exception', or which entirely discards the cause.

Reply:
Right. If that ever happens, we can adjust accordingly.